ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/credentials/services/CredentialsClientService.kt

```diff
@@ -28,14 +29,16 @@ import com.izivia.ocpi.toolkit.transport.TransportClientBuilder
  * @property clientCredentialsRoleRepository client's repository to retrieve its role
  * @property serverVersionsEndpointUrl the versions endpoint url of the server (for the client to retrieve endpoints)
  * @property transportClientBuilder used to build a transport (will be used to create CredentialClient to make calls)
+ * @property requiredOtherPartEndpointsProvider the endpoints this client expects from the other part to provide
  */
 class CredentialsClientService(
     private val clientVersionsEndpointUrl: String,
     private val clientPartnerRepository: PartnerRepository,
     private val clientVersionsRepository: VersionsRepository,
     private val clientCredentialsRoleRepository: CredentialsRoleRepository,
     private val serverVersionsEndpointUrl: String,
-    private val transportClientBuilder: TransportClientBuilder
+    private val transportClientBuilder: TransportClientBuilder,
+    private val requiredOtherPartEndpointsProvider: suspend () -> Map<InterfaceRole, List<ModuleID>>
```

Comment:
is there a particular reason why need this as a function? instead of just a map? ... I'd assume this will be static anyways, as your particular implementation will require certain endpoints, or not

nit: name could simply be `requiredEndpoints` as it should be pretty clear we are testing the other side, not ourself